### ISSUE TITLE
Training restarts at every epoch

### ISSUE THREAD

**Issue by hubbup** [tensorflow/models#3568]
I created two tfrecord files (for training and validation) which have two-class labels and images.
Then, I modified "models/official/resnet/imagenet_main.py".

> INFO:tensorflow:learning_rate = 0.0125, cross_entropy = 0.00017373938, train_accuracy = 0.5 (8.141 sec)
INFO:tensorflow:loss = 0.29520494, step = 11850 (8.141 sec)
INFO:tensorflow:global_step/sec: 12.8693
INFO:tensorflow:learning_rate = 0.0125, cross_entropy = 5.276655e-05, train_accuracy = 0.6666667 (7.770 sec)
INFO:tensorflow:loss = 0.29434904, step = 11950 (7.770 sec)
INFO:tensorflow:global_step/sec: 12.8106
INFO:tensorflow:learning_rate = 0.0125, cross_entropy = 5.574875e-05, train_accuracy = 0.75 (7.806 sec)
INFO:tensorflow:loss = 0.29361826, step = 12050 (7.806 sec)
INFO:tensorflow:global_step/sec: 12.7354
INFO:tensorflow:learning_rate = 0.0125, cross_entropy = 0.00017654907, train_accuracy = 0.8 (7.853 sec)
INFO:tensorflow:loss = 0.29300678, step = 12150 (7.852 sec)

It trains well on the first epoch, but it seems that the weight values are initialized after the first epoch.
When it started the 2nd epoch, the accuracy went to "zero" again, even though the saved model was restored. This kind of problems are repeated at every epoch.

Can you guess the reason?

(additional question) anyone knows why the first accuracy value is "zero"?

> INFO:tensorflow:Starting evaluation at 2018-03-12-09:53:24
INFO:tensorflow:Restoring parameters from /mnt/data/AI/vi/98_study/tensorflow/models/official/resnet/resnet_model/model.ckpt-14686
INFO:tensorflow:Finished evaluation at 2018-03-12-09:53:56
INFO:tensorflow:Saving dict for global step 14686: accuracy = 0.3605959, global_step = 14686, loss = 5.485213
{'loss': 5.485213, 'global_step': 14686, 'accuracy': 0.3605959}
Starting a training cycle.
INFO:tensorflow:Create CheckpointSaverHook.
INFO:tensorflow:Restoring parameters from /mnt/data/AI/vi/98_study/tensorflow/models/official/resnet/resnet_model/model.ckpt-14686
INFO:tensorflow:Saving checkpoints for 14687 into /mnt/data/AI/vi/98_study/tensorflow/models/official/resnet/resnet_model/model.ckpt.
INFO:tensorflow:learning_rate = 0.0125, cross_entropy = 7.5467443, train_accuracy = 0.0
INFO:tensorflow:loss = 7.821854, step = 14687
INFO:tensorflow:global_step/sec: 12.3532
INFO:tensorflow:learning_rate = 0.0125, cross_entropy = 0.003049913, train_accuracy = 0.5 (8.095 sec)
INFO:tensorflow:loss = 0.27778482, step = 14787 (8.095 sec)
INFO:tensorflow:global_step/sec: 12.9102
INFO:tensorflow:learning_rate = 0.0125, cross_entropy = 0.00045617242, train_accuracy = 0.6666667 (7.746 sec)
INFO:tensorflow:loss = 0.27451897, step = 14887 (7.746 sec)
INFO:tensorflow:global_step/sec: 12.9047
INFO:tensorflow:learning_rate = 0.0125, cross_entropy = 0.0003769689, train_accuracy = 0.75 (7.749 sec)
INFO:tensorflow:loss = 0.27376384, step = 14987 (7.749 sec)

**Comment by hubbup** [comment]
Solved. It was the problem of unsorted dataset.

**Comment by martingra** [comment]
What do you mean by “unsorted dataset”? Do you mean that the labels were not correctly matched with the data?
